### PR TITLE
FINERACT-1109 appropriate-interest-larger-than-emi

### PR DESCRIPTION
**Pull request by fynmanoj**
## Description
Describe the changes made and why they were made. Ignore if these details are present on the associated Jira ticket
https://mifosforge.jira.com/browse/AL-14
## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [x] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [ ] API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm has been updated with details of any API changes.

- [x] Integration tests have been created/updated for verifying the changes made.

- [x] All Integrations tests are passing with the new commits.

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

**Comment by xurror** [comment]
You got checkstyle violations. Run `./gradlew build` and  `./gradlew check` and make sure everything works fine.

**Comment by vorburger** [comment]
@avikganguly01 knowing that @fynmanoj works with you, and assuming that you are still an active committer to this project (are you?), I was wondering if you would like to engage on code review feedback and possible evental merge of PRs such as this one? (I don't want to be a bottleneck on this project, and won't have the spare time to help review changes like this. If there are other active commiters such as @awasum @ptuomola @xurror or others reading this who want to engage on this PR, please don't let me hold you up / wait for me - I'll ignore this one, and try to help on others.)

**Comment by vorburger** [comment]
@fynmanoj this PR needs to have a JIRA issue on the ASF JIRA, instead of referencing an issue on https://mifosforge.jira.com.

**Comment by edcable** [comment]
@vorburger I opened up tickets on JIRA for these pull requests and referenced them in the title of the PR. I tagged @avikganguly01 as a reviewer and assume he can do the needful in reviewing and merging the PRs as we are going to start testing this development branch with these pull requests merged on the user's test server. 

@fynmanoj we also have to address the failing build for this PR.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/domain/ConfigurationDomainServiceJpa.java
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/domain/ConfigurationDomainServiceJpa.java:258]: This should have been part of PR FIN-1106.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java:286]: Is tranche loan approval failing due to flag?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DecliningBalanceInterestLoanScheduleGenerator.java
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/DecliningBalanceInterestLoanScheduleGenerator.java:155]: LGTM

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelRepaymentPeriod.java
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelRepaymentPeriod.java:157]: Looks like a different bug got squashed here. What effects could this bug have been producing? Shall we create a different ticket to track this?

**Review by avikganguly01 — CHANGES_REQUESTED** [review]
Rebase required. Missing integration test for this scenario.

https://travis-ci.org/github/apache/fineract/builds/713781505 - Tranche loan creation is failing. Can you check if it's a valid regression?

Have you tested same scenario for a new loan where there is a long gap between disbursement and first repayment date? If this is resolved, we can create and mention it in a separate bug blocked by resolution of this issue.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelRepaymentPeriod.java
Comment (fynmanoj) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelRepaymentPeriod.java:157]: yes, found this while testing the new feature.

**Comment by edcable** [comment]
@avikganguly01 Did the recent changes made by @fynmanoj address the feedback you provided?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelRepaymentPeriod.java
Comment (fynmanoj) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelRepaymentPeriod.java:157]: In the add Interest method principle was added to the total due instead of the interest. which is wrong, as the added interest must reflect in the total due.

**Pull request merged**
